codeface_extraction/extractions.py

```diff
@@ -266,6 +266,47 @@ def __init__(self, dbm, conf, resdir, csv_writer):
                     # LIMIT 10
                 """
 
+# Extraction of function implementations
+class FunctionImplementationExtraction(Extraction):
+    def __init__(self, dbm, conf, resdir, csv_writer):
+        Extraction.__init__(self, dbm, conf, resdir, csv_writer)
+
+        self.file_name = "implementations.list"
+
+        # for subclasses
+        self.sql = """
+                    SELECT c.id,
+                           c.commitHash, c.ChangedFiles,
+                           cd.file, cd.entityId, cd.impl
+
+                    FROM project p
+
+                    # get commits for project
+                    JOIN commit c ON p.id = c.projectId
+
+                    # get commit meta-data
+                    LEFT JOIN commit_dependency cd ON c.id = cd.commitId
+
+                    # filter for current project
+                    WHERE p.name = '{project}'
+                    AND p.analysisMethod = '{tagging}'
+                    AND cd.file IS NOT NULL
+
+                    ORDER BY cd.file, c.id, c.commitHash, cd.entityId
```

Comment:
Wouldn't it be better for debugging to have another sorting here? E.g., `ORDER BY c.id, c.commitHash, cd.file, cd.entityId` or `ORDER BY cd.file, cd.entityId, c.id, c.commitHash`. In the end, my suggestion is to sort by (not in the SQL  sense!) file *and* function name.